executor/executor_xapi.go

```diff
@@ -118,8 +119,13 @@ func (e *XSelectTableExec) doRequest() error {
 	selReq.Fields = resultFieldsToPBExpression(e.tablePlan.Fields())
 	selReq.Where = e.where
 	selReq.Ranges = tableRangesToPBRanges(e.tablePlan.Ranges)
-	selReq.Limit = e.tablePlan.LimitCount
-
+	if e.supportDesc {
+		if e.tablePlan.Desc {
+			selReq.OrderBy = append(selReq.OrderBy, &tipb.ByItem{Desc: &e.tablePlan.Desc})
+		}
+		// Limit can be pushed only if desc is supported.
```

Comment:
This will be simpler, we don't care if the scan is descending, we can just push limit.